src/browser/index.jsx

```diff
@@ -15,22 +14,47 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import {remote, ipcRenderer} from 'electron';
 
-import buildConfig from '../common/config/buildConfig';
-import settings from '../common/settings';
 import utils from '../utils/util';
 
+import Config from '../common/config';
+
 import MainPage from './components/MainPage.jsx';
-import AppConfig from './config/AppConfig.js';
 import {createDataURL as createBadgeDataURL} from './js/badge';
 
-const teams = settings.mergeDefaultTeams(AppConfig.data.teams);
+const config = new Config(remote.app.getPath('userData') + '/config.json');
+
+const teams = config.teams;
 
 remote.getCurrentWindow().removeAllListeners('focus');
 
 if (teams.length === 0) {
   remote.getCurrentWindow().loadFile('browser/settings.html');
 }
 
+const permissionRequestQueue = [];
+const requestingPermission = new Array(teams.length);
+
+const parsedURL = url.parse(window.location.href, true);
+const initialIndex = parsedURL.query.index ? parseInt(parsedURL.query.index, 10) : 0;
+
+let deeplinkingUrl = null;
+if (!parsedURL.query.index || parsedURL.query.index === null) {
+  deeplinkingUrl = remote.getCurrentWindow().deeplinkingUrl;
+}
+
+config.on('update', (configData) => {
+  teams.splice(0, teams.length, ...configData.teams);
+  requestingPermission.length = teams.length;
+});
+
+config.on('synchronize', () => {
+  ipcRenderer.send('reload-config');
```

Comment:
Maybe I’m not following cause this is the second time, is this event being handled somewhere else and doing something else apart from config.reload()?

Reply:
See above. :)